### ISSUE TITLE
Enhancement: Forbid duplicated accessibility modifiers

### ISSUE THREAD

**Issue by fisker** [typescript-eslint/typescript-eslint#7495]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

Forbid duplicated accessibility modifiers.

### Fail

```typescript
class Foo {
  public public a;
  private public b;
  protected private c;
  public protected d;
  public protected private e;
}
```

### Pass

```typescript
Should throw errors.
```
